Felienne/hedy
# [CHORE] Cleaned up auth.ts, re-wrote routes to the back-end

**Description**
In this PR we try to take a first step into better dividing the functionality of both the back-end and the front-end. Currently `auth.ts` handles a lot of functionality that, in my opinion, should be handled in the back-end. Such as re-directing pages when already logged in, storing GET attributes etc. We re-write the structure of `auth.ts` to only focus on handling the front-end relevant stuff. We also fix some missing error templates, small styling issues and replace the old error modal with our new one.

**Fixes**
This PR fixes #2089 and fixes #2099

**How to test**
This is a clean-up task, if nothing is broken it works. One small think that can be tested is to verify that you are not allowed to visit the `/reset` page when being logged in. You will simply be re-directed to your profile.

**Checklist**
Done? Check if you have it all in place using this list*
  

* If you're unsure about any of these, don't hesitate to ask. We're here to help!
